examples/adder/cmd.go

```diff
@@ -29,36 +29,37 @@ var RootCmd = &cmds.Command{
 			Arguments: []cmdkit.Argument{
 				cmdkit.StringArg("summands", true, true, "values that are supposed to be summed"),
 			},
-			Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) {
+			Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
 				sum := 0
 
 				for i, str := range req.Arguments {
 					num, err := strconv.Atoi(str)
 					if err != nil {
 						re.SetError(err, cmdkit.ErrNormal)
-						return
+						return err
 					}
 
 					sum += num
 					re.Emit(fmt.Sprintf("intermediate result: %d; %d left", sum, len(req.Arguments)-i-1))
 				}
 
 				re.Emit(fmt.Sprintf("total: %d", sum))
+				return nil
 			},
 		},
 		// a bit more sophisticated
 		"encodeAdd": &cmds.Command{
 			Arguments: []cmdkit.Argument{
 				cmdkit.StringArg("summands", true, true, "values that are supposed to be summed"),
 			},
-			Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) {
+			Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
 				sum := 0
 
 				for i, str := range req.Arguments {
 					num, err := strconv.Atoi(str)
 					if err != nil {
 						re.SetError(err, cmdkit.ErrNormal)
-						return
+						return err
```

Comment:
shouldn't this remove the above `SetError` call, as otherwise these get emitted twice?

Reply:
Good catch. I'll fix those.